.github/workflows/release.yml

```diff
@@ -0,0 +1,145 @@
+jobs:
+  release:
+    if: ${{ github.event.inputs.confirm == 'true' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Check if user is in group"
+        run: |
+          MEMBER=$( curl -s \
+                 --request GET \
+                 --header 'authorization: Bearer ${{ secrets.GITTOKEN }}' \
+                 --header 'content-type: application/json' \
+                 --url https://api.github.com/orgs/matomo-org/teams/release-team/members \
+                 | jq '.[] | select(.login == "${{ github.actor }}")' )
+
+          if [[ -z "$MEMBER" ]]
+          then
+           echo "Action was not triggered by a member with release permission"
+           exit 1;
+          fi
+        shell: bash
+      - uses: actions/checkout@v2
+        with:
+          lfs: false
+      - name: Import GPG key
+        id: import_gpg
+        run: |
+          echo "${{ secrets.GPG_CERTIFICATE }}" > $HOME/private.asc
+          gpg --import --batch --yes $HOME/private.asc
+          echo "default-cache-ttl 7200
+          max-cache-ttl 31536000
+          allow-preset-passphrase" > $HOME/.gnupg/gpg-agent.conf
+          keygrip=$(gpg --import --import-options show-only --with-keygrip $HOME/private.asc | grep "Keygrip" | grep -oP "([A-F0-9]+)" | head -1)
+          hexPassphrase=$( echo -n "${{ secrets.GPG_CERTIFICATE_PASS }}" | od -A n -t x1 -w100 | sed 's/ *//g' )
+          gpg-connect-agent "RELOADAGENT" /bye
+          gpg-connect-agent "PRESET_PASSPHRASE ${keygrip} -1 ${hexPassphrase}" /bye
+          gpg-connect-agent "KEYINFO ${keygrip}" /bye
+      - name: Check preconditions, create tag, build and publish release
+        id: tag
+        run: |
+          if [[ -n "${{ github.event.inputs.version }}" ]]
+          then
+            version="${{ github.event.inputs.version }}"
+            echo "Version to re-build: '$version'"
+
+            git fetch --tags -q 2>/dev/null
+            tag_exists=$( git tag --list "$version" )
+
+            if [[ -z "$tag_exists" ]]
+            then
+              echo "A tag for $version does not exist."
+              exit 1
+            fi
+
+            echo ::set-output name=update::true
+          else
+            version=$( cat core/Version.php | grep -oP "VERSION = '\K([^\']+)" )
+            echo "Version to build: '$version'"
+
+            git fetch --tags -q 2>/dev/null
+            tag_exists=$( git tag --list "$version" )
+
+            if [[ -n "$tag_exists" ]]
+            then
+              echo "A tag for $tag_exists already exists."
+              exit 1
+            fi
+
+            if [[ ${GITHUB_REF#refs/heads/} != "4.x-dev" && ${GITHUB_REF#refs/heads/} != "next_release" ]]
+            then
+              echo "A tag can only be created from branches '4.x-dev' and 'next_release'. Please create the tag manually if a release needs to be built from another branch."
+              exit 1
+            fi
+
+            echo "Creating a tag for $version"
```

Comment:
can we add a check if released from `*.x-dev` branch that this is only allowed for alpha/beta releases?

Reply:
Ok. Only beta release tags are now allowed to be created from `*.x-dev` branches within the action.